src/components/PeoplePage.tsx

```diff
@@ -59,38 +59,33 @@ export const PeoplePage = () => {
     });
   }, [people, searchParams]);
 
-  const getPeopleWithParents = () => {
-    setPeople((prev) => {
-      if (!prev) {
-        return null;
-      }
-
-      return prev.map((person) => {
-        return {
-          ...person,
-          mother: prev.find((mother) => mother.name === person.motherName),
-          father: prev.find((father) => father.name === person.fatherName),
-        };
-      });
+  const getPeopleWithParents = (prev: Person[]) => {
+    return prev.map((person) => {
+      return {
+        ...person,
+        mother: prev.find((mother) => mother.name === person.motherName),
+        father: prev.find((father) => father.name === person.fatherName),
+      };
     });
   };
 
-  const getPeopleList = () => {
-    getPeople()
-      .then((result) => setPeople(result))
-      .then(() => getPeopleWithParents())
-      .catch(() => setIsLoadingError(true))
-      .finally(() => setIsLoaded(true));
+  const getPeopleList = async () => {
+    try {
+      const peopleList = await getPeople();
+      const peopleWithParents = getPeopleWithParents(peopleList);
+
+      setPeople(peopleWithParents);
+    } catch {
+      setIsLoadingError(true);
+    } finally {
+      setIsLoaded(true);
```

Comment:
It's better to set setIsLoader(true) before try { ... } and set false in finally { setIsLoader(false) }